homeassistant/components/overkiz/config_flow.py

```diff
@@ -1,78 +1,203 @@
+            )
+
+            if developer_mode is None:
+                raise DeveloperModeDisabled
+
+            token = await client.generate_local_token(gateway_id)
+            await client.activate_local_token(
+                gateway_id=gateway_id, token=token, label="Home Assistant/local"
+            )
+
+            # Verify SSL blocked by https://github.com/Somfy-Developer/Somfy-TaHoma-Developer-Mode/issues/5
+            # Somfy (self-signed) SSL cert uses the wrong common name
+            session = async_create_clientsession(self.hass, verify_ssl=False)
+
+            # Local API
+            local_client = OverkizClient(
+                username="",
+                password="",
+                token=token,
+                session=session,
+                server=generate_local_server(host=user_input[CONF_HOST]),
+            )
+
+            await local_client.login()
+
+            user_input[CONF_TOKEN] = token
+        else:
+            server = SUPPORTED_SERVERS[user_input[CONF_SERVER]]
+            client = self._create_cloud_client(
+                username=username, password=password, server=server
+            )
+
+            await client.login(register_event_listener=False)
+
+        # Set main gateway id as unique id
         if gateways := await client.get_gateways():
-            gateway_id = gateways[0].id
-            await self.async_set_unique_id(gateway_id)
+            for gateway in gateways:
+                if is_overkiz_gateway(gateway.id):
+                    gateway_id = gateway.id
+                    await self.async_set_unique_id(gateway_id)
+
+        return user_input
 
     async def async_step_user(
         self, user_input: dict[str, Any] | None = None
     ) -> FlowResult:
         """Handle the initial step via config flow."""
-        errors = {}
+        if user_input:
+            self._server = user_input[CONF_SERVER]
+
+            # Some Overkiz hubs do support a local API
+            # Users can choose between local or cloud API.
+            if self._server in SERVERS_WITH_LOCAL_API:
+                return await self.async_step_local_or_cloud()
+
+            return await self.async_step_cloud()
+
+        return self.async_show_form(
+            step_id="user",
+            data_schema=vol.Schema(
+                {
+                    vol.Required(CONF_SERVER, default=self._server): vol.In(
+                        {key: hub.name for key, hub in SUPPORTED_SERVERS.items()}
+                    ),
+                }
+            ),
+        )
+
+    async def async_step_local_or_cloud(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Users can choose between local API or cloud API via config flow."""
+        if user_input:
+            self._api_type = user_input[CONF_API_TYPE]
+
+            if self._api_type == APIType.LOCAL:
+                return await self.async_step_local()
+
+            return await self.async_step_cloud()
```

Comment:
Can we not offer local if we know the device doesn't support it?

Reply:
What do you mean? For Overkiz there are multiple hubs with corresponding cloud servers. Only some Somfy hubs do support the local API.